### PR TITLE
Add docstring for `percival`

### PR DESCRIPTION
**Pull request by tmigot**
Following JSO standard https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/main/src/R2.jl

### CODE REVIEW

File: src/method.jl [hunk 1]
```diff
@@ -59,22 +60,54 @@ end
...
-- inity: Initial values of the Lagrangian multipliers
-- subsolver_kwargs: subsolver keyword arguments as a dictionary
-- `verbose::Integer = 0`: if > 0, display iteration details every `verbose` iteration.
+# Arguments
+- `nlp::AbstractNLPModel{T, V}` is the model to solve, see `NLPModels.jl`.
+
+# Keyword arguments 
+- `x::V = nlp.meta.x0`: the initial guess.
+- `atol::T = T(1e-8)`: absolute tolerance.
+- `rtol::T = T(1e-8)`: relative tolerance: algorithm stops when ‖∇f(xᵏ)‖ ≤ atol + rtol * ‖∇f(x⁰)‖.
```
Comment (dpo) [inline src/method.jl:77]: There are constraints, so this stopping test isn't right.

Comment (tmigot) [reply src/method.jl]: Thanks for the comment. It is fixed.

File: src/method.jl [hunk 2]
```diff
@@ -73,16 +73,16 @@ For advanced usage, first define a `PercivalSolver` to preallocate the memory us
...
-- `max_eval::Int = 100000`: maximum number of evaluation of the objective function.
-- `max_time::Float64 = 30.0`: maximum time limit in seconds.
-- `max_iter::Int = 2000`: maximum number of iterations.
-- `verbose::Int = 0`: if > 0, display iteration details every `verbose` iteration.
-- `μ::Real = T(10.0)`: Starting value of the penalty parameter.
-- `subsolver_logger::AbstractLogger = NullLogger()`: logger passed to `tron`.
-- `inity = nothing`: initial values of the Lagrangian multipliers. If `nothing` the algorithm uses `Krylov.cgls` to compute an approximation.
+- `atol::T = T(1e-8)`: absolute tolerance;
+- `rtol::T = T(1e-8)`: relative tolerance;
+- `ctol::T = T(1e-8)`: absolute tolerance on the feasibility: algorithm stops when ‖∇f(xᵏ)‖ ≤ atol + rtol * ‖∇f(x⁰)‖ and ‖c(xᵏ)‖ ≤ ctol;
```
Comment (dpo) [inline src/method.jl:78]: No, I don't think that's right either. What about the multipliers?

Comment (tmigot) [reply src/method.jl]: Oh yes, sorry. I added this https://github.com/JuliaSmoothOptimizers/Percival.jl/blob/b24c3429fd71aad1f213d1f4f6190dfccc5f9960/src/method.jl#L88

**Review by dpo — APPROVED** [review]

**Pull request merged**